### PR TITLE
Added context utilities to the base-MEF

### PR DESCRIPTION
**Pull request by Dtphelan1**
# Summary

After realizing how many extractors will depend on getting the patient from context, this PR moves the utility from the E-MEF to this repo.  

## New behavior

Minimal, as this PR doesn't concern itself with using the functionality anywhere in this repo. 

## Code changes

- Adds this utility;
- Adds tests for the utility;
- Exports the utility.

# Testing guidance

- Make sure tests pass, especially the new ones. 
- Connect this with the E-MEF repo and ensure that tests pass there too.

### CODE REVIEW

**Comment by jafeltra** [comment]
This looks good! I had one question about the message of the Error over in E-MEF, so I wanted to wait to hear your opinion on that before approving this PR. But besides that, it looks good to me!

File: src/helpers/contextUtils.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const logger = require('./logger');
+const { getBundleResourcesByType } = require('./fhirUtils');
+
+async function getPatientFromContext(mrn, context) {
+  const patientInContext = getBundleResourcesByType(context, 'Patient', {}, true);
+  if (!patientInContext) {
+    throw Error('Could not find a patient in context');
```
Comment (Dtphelan1) [inline src/helpers/contextUtils.js:7]:
```suggestion
    throw Error('Could not find a patient in context; ensure that a PatientExtractor is used earlier in your extraction configuration');
```

File: test/helpers/contextUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      },
+    ],
+  };
+  test('Should return Patient resource in context', async () => {
+    const patient = await getPatientFromContext(MOCK_PATIENT_MRN, patientContext);
+    expect(patient.id).toEqual(patientResource.id);
+  });
+
+  test('Should throw an error if there is no patient in context', async () => {
+    await expect(getPatientFromContext(MOCK_PATIENT_MRN, {})).rejects.toThrow('Could not find a patient in context');
```
Comment (Dtphelan1) [inline test/helpers/contextUtils.test.js:26]:
```suggestion
    await expect(getPatientFromContext(MOCK_PATIENT_MRN, {})).rejects.toThrow('Could not find a patient in context; ensure that a PatientExtractor is used earlier in your extraction configuration');
```

File: src/helpers/contextUtils.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const logger = require('./logger');
+const { getBundleResourcesByType } = require('./fhirUtils');
+
+async function getPatientFromContext(mrn, context) {
+  const patientInContext = getBundleResourcesByType(context, 'Patient', {}, true);
```
Comment (Dtphelan1) [inline src/helpers/contextUtils.js:5]:
```suggestion
  logger.debug('Getting patient from context');
  const patientInContext = getBundleResourcesByType(context, 'Patient', {}, true);
```

**Review by jafeltra — APPROVED** [review]
New `logger.debug` and error message look good to me!

**Review by julianxcarter — APPROVED** [review]
This looks great to me, and worked just as well when I tied it in to the epic MEF. This is the first of two approvals, nice job! @Dtphelan1

**Pull request merged**